test/jdk/java/util/Currency/ValidateISO4217.java

```diff
@@ -28,285 +29,293 @@
-
-        /* Not defined in ISO 4217 list, but defined in .properties file. */
-        {"AX", "EUR", "978", "2"},      // \u00c5LAND ISLANDS
-        {"PS", "ILS", "376", "2"},      // Palestinian Territory, Occupied
-
-        /* Not defined in ISO 4217 list, but added in ISO 3166 country code list */
-        {"JE", "GBP", "826", "2"},      // Jersey
-        {"GG", "GBP", "826", "2"},      // Guernsey
-        {"IM", "GBP", "826", "2"},      // Isle of Man
-        {"BL", "EUR", "978", "2"},      // Saint Barthelemy
-        {"MF", "EUR", "978", "2"},      // Saint Martin
-
-        /* Defined neither in ISO 4217 nor ISO 3166 list */
-        {"XK", "EUR", "978", "2"},      // Kosovo
-    };
-
-    /* Codes that are obsolete, do not have related country, extra currency */
-    static final String otherCodes =
-        "ADP-AFA-ATS-AYM-AZM-BEF-BGL-BOV-BYB-BYR-CHE-CHW-CLF-COU-CUC-CYP-"
-        + "DEM-EEK-ESP-FIM-FRF-GHC-GRD-GWP-IEP-ITL-LTL-LUF-LVL-MGF-MRO-MTL-MXV-MZM-NLG-"
-        + "PTE-ROL-RUR-SDD-SIT-SLL-SKK-SRG-STD-TMM-TPE-TRL-VEF-UYI-USN-USS-VEB-VED-"
-        + "XAG-XAU-XBA-XBB-XBC-XBD-XDR-XFO-XFU-XPD-XPT-XSU-XTS-XUA-XXX-"
-        + "YUM-ZMK-ZWD-ZWN-ZWR";
-
-    static boolean err = false;
-
-    static Set<Currency> testCurrencies = new HashSet<Currency>();
-
-    public static void main(String[] args) throws Exception {
-        CheckDataVersion.check();
-        test1();
-        test2();
-        getAvailableCurrenciesTest();
-
-        if (err) {
-            throw new RuntimeException("Failed: Validation ISO 4217 data");
-        }
+    // Input golden-data file
+    private static final File dataFile = new File(System.getProperty(
+            "test.src", "."), "tablea1.txt");
+    // Code statuses
+    private static final byte UNDEFINED = 0;
+    private static final byte DEFINED = 1;
+    private static final byte SKIPPED = 2;
+    private static final byte TESTED = 4;
+    private static final int ALPHA_NUM = 26;
+    // An alpha2 code table which maps the status of a country
+    private static final byte[] codes = new byte[ALPHA_NUM * ALPHA_NUM];
+    // Codes derived from ISO4217 golden-data file
+    private static final List<Arguments> ISO4217Codes = new ArrayList<Arguments>();
+    // Additional codes not from the ISO4217 golden-data file
+    private static final List<Arguments> additionalCodes = new ArrayList<Arguments>();
+    // Currencies to test (derived from ISO4217Codes and additionalCodes)
+    private static final Set<Currency> testCurrencies = new HashSet<>();
+    private static SimpleDateFormat format = null;
+
+    // Sets up the following test data:
+    // ISO4217Codes, additionalCodes, testCurrencies, codes
+    @BeforeAll
+    static void setUpTestingData() throws Exception {
+        // These functions laterally setup 'testCurrencies' and 'codes'
+        // at the same time
+        setUpISO4217Codes();
+        setUpAdditionalCodes();
+        finishTestCurrencies();
```

Comment:
The method name `'finish` is kind of confusing. I'd explicitly describe what the method does, i.e, setup `other` currencies.

Reply:
Renamed it to your suggestion